consensus/quorum/one-node-staked-vote.go

```diff
@@ -39,31 +45,52 @@ func (v *stakedVoteWeight) Policy() Policy {
 
 // IsQuorumAchieved ..
 func (v *stakedVoteWeight) IsQuorumAchieved(p Phase) bool {
-	// TODO Implement this logic w/Chao
-	// soFar := numeric.ZeroDec()
+	t := numeric.NewDecFromBigInt(v.QuorumThreshold())
+	currentTotalPower := v.computeCurrentTotalPower(p)
+
+	utils.Logger().Info().
+		Str("policy", v.Policy().String()).
+		Str("phase", p.String()).
+		Str("threshold", t.String()).
+		Str("total-power-of-signers", currentTotalPower.String()).
+		Msg("Attempt to reach quorum")
+
+	return currentTotalPower.GT(t)
+}
+
+func (v *stakedVoteWeight) computeCurrentTotalPower(p Phase) numeric.Dec {
 	w := shard.BlsPublicKey{}
 	members := v.Participants()
+	currentTotalPower := numeric.ZeroDec()
+	ourCount := numeric.NewDecFromBigInt(big.NewInt(v.hmySlotCount))
 
 	for i := range members {
-		w.FromLibBLSPublicKey(members[i])
-		// isHMY := v.validatorStakes[w].isHarmonyNode
-		if v.ReadSignature(p, members[i]) == nil {
-			// TODO TODO finish this logic
+		if v.ReadSignature(p, members[i]) != nil {
+			w.FromLibBLSPublicKey(members[i])
+			if v.validatorStakes[w].isHarmonyNode {
+				currentTotalPower = currentTotalPower.Add(
+					two.Mul(v.stakedTotal).Add(one).Quo(ourCount),
```

Comment:
Use hard coded voting power percentage for harmony (68%) and staker (32%)

Reply:
yep - mostly done with the refactor that way already - will finish in early morning before you're in office 👍

Reply:
Cool. Tomorrow, I may not be in office as my kid is sick again with fever and cough. I will see tomorrow morning his condition.